store/types/validity.go

```diff
@@ -1,15 +1,28 @@
 package types
 
-// AssertValidKey checks if the key is valid(key is not nil)
+const (
+	// 128K - 1
+	MaxKeyLength = (1 << 17) - 1
```

Comment:
I would have preferred these two constants to be variables that can be modified at startup, I think the key-lengths are reasonable but we don't have full knowledge of what other protocols are doing ( I doubt even 'them' have no clear idea of the size of their own keys)

EG in `cmd/appd/main.go`:
```golang

func main() {
       storetypes.MaxKeyLength = my safe limit
       storetypes.MaxValueLength = my safe limit
       
       rootCmd().Execute ....
}
```

Reply:
Sounds reasonable to me. It would allow new chains to be much more restrictive on key lengths for example. Existing chains can scan their keys in usage and put a limit based on their real world data.

Reply:
sounds good, done.